test/test_jit_py3.py

```diff
@@ -108,6 +110,44 @@ def foo(x):
         self.assertEqual(out.sequence_features, [3.0])
         self.assertEqual(out.time_since_first, 3.0)
 
+    def test_types_as_values(self):
+        def fn(m: torch.Tensor) -> torch.device:
+            return m.device
+
+        self.checkScript(fn, [torch.randn(2, 2)])
+
+        GG = namedtuple('GG', ['f', 'g'])
+
+        class Foo(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+
+            @torch.jit.ignore
+            def foo(self, x, z):
```

Comment:
I believe this already works on master - @ignored` functions resolve types using the python resolver. Could you add tests not using `@ignored`. ?